### DISCUSSION TITLE
Logging 5xx errors

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10319] [Q&A]
Hi, I use https://github.com/KyleRoss/node-lambda-log to log in my serverless function in a very standard **aws-node-project** project but when there is a 5xx, sometimes I somehow don't have a corresponding error log.

Am I missing a trick to log the response in serverless.yml? Thank you!

**Comment by pgrzesik** [comment]
Hey @kaihendry - could you provide a simple reproducible example on when that happens?

**Reply by kaihendry** [reply]
I'm unable to reproduce it. I now know it's a 504 timeout.

**Reply by pgrzesik** [reply]
Maybe in that situation that log is not able to be emitted because it's supposed to be logged after regular Lambda timeout?

**Reply by kaihendry** [reply]
AWS has responded to me saying that I should pass the request ID to the function via a API gateway mapping? https://aws.amazon.com/blogs/compute/techniques-and-tools-for-better-serverless-api-logging-with-amazon-api-gateway-and-aws-lambda/

I'm struggling to figure this out....

    events:
      - http:
          path: v1/document
          method: post
          request:
            headers:
              apigw-request-id:
                mappedValue: $context.requestId

I'm looking for apigw-request-id in the headers, but can't find it!

**Reply by pgrzesik** [reply]
Hello @kaihendry - did you manage to figure this one out?

**Reply by kaihendry** [reply]
Tbh the break through to help me track this problem was log insights queries like:

    fields @timestamp, @message | sort @timestamp asc | filter @message like "Task timed out"

and

```
fields @timestamp, @message, @duration, @requestId
| filter @duration > 100
```
